### PR TITLE
26 frontend structure + login + register

### PR DESCRIPTION
**Pull request by fabianmueller7**

### CODE REVIEW

**Review by maxdeans — DISMISSED** [review]

**Review by maxdeans — DISMISSED** [review]

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,12 +1,48 @@
...
   { path: 'home', component: HomeComponent },
   { path: 'design', component: UiTestComponent },
+  { path: 'login', component: LoginComponent },
+  {
+    path: 'storify',
+    component: StorifyComponent,
+    children: [
+      {
+        path: '',
+        component: ShomeComponent,
```
Comment (DawidKapka) [inline src/app/app-routing.module.ts:25]: redirectTo

File: src/app/register/register.component.html [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      type="password"
+      id="password"
+      name="password"
+      placeholder="password"
+      #password
+    ></app-textfield>
+    <br />
+  </div>
+  <!-- TODO: Improve this-->
+  <div *ngIf="isLoading">LOADING...</div>
```
Comment (DawidKapka) [inline src/app/register/register.component.html:37]: implement loading animation (loader-component)

File: src/app/storify/create/create.component.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+@Component({
+  selector: 'app-create',
+  templateUrl: './create.component.html',
+  styleUrls: ['./create.component.scss'],
+})
+export class CreateComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
+}
```
Comment (DawidKapka) [inline src/app/storify/create/create.component.ts:12]: empty constructor and ngOnInit -> remove

File: src/app/storify/game/game.component.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+@Component({
+  selector: 'app-game',
+  templateUrl: './game.component.html',
+  styleUrls: ['./game.component.scss'],
+})
+export class GameComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
+}
```
Comment (DawidKapka) [inline src/app/storify/game/game.component.ts:12]: again empty constructor and ngOnInit -> remove

File: src/app/storify/join/join.component.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+@Component({
+  selector: 'app-join',
+  templateUrl: './join.component.html',
+  styleUrls: ['./join.component.scss'],
+})
+export class JoinComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
+}
```
Comment (DawidKapka) [inline src/app/storify/join/join.component.ts:12]: dito: empty constructor & ngOnInit

File: src/app/storify/shome/shome.component.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+@Component({
+  selector: 'app-shome',
+  templateUrl: './shome.component.html',
+  styleUrls: ['./shome.component.scss'],
+})
+export class ShomeComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
+}
```
Comment (DawidKapka) [inline src/app/storify/shome/shome.component.ts:12]: again

File: src/app/storify/storify.component.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+@Component({
+  selector: 'app-storify',
+  templateUrl: './storify.component.html',
+  styleUrls: ['./storify.component.scss'],
+})
+export class StorifyComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
+}
```
Comment (DawidKapka) [inline src/app/storify/storify.component.ts:12]: again...

File: src/styles.scss [hunk 8]
```diff
@@ -3,3 +3,58 @@
...
+    font-family: "Silkscreen", cursive;
+    margin: 1;
+    white-space: normal;
+    display: block;
+  }
+
+  .link {
+    margin-bottom: 0px;
+  }
+}
```
Comment (DawidKapka) [inline src/styles.scss:60]: component-specific styles should not be added to global styles.scss -> move to specific components

File: src/app/register/register.component.html [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  <br />
+  <div class="label">
+    <label class="text" for="email">Email:</label><br />
+    <app-textfield type="email" id="email" name="email" placeholder="email" #email></app-textfield>
+  </div>
+  <br />
+  <div class="label">
+    <label class="text" for="password">Password:</label><br />
+    <app-textfield type="password" id="password" name="password" placeholder="password" #password></app-textfield>
+    <br />
```
Comment (DawidKapka) [inline src/app/register/register.component.html:16]: hide content while loading

File: src/app/storify/create/create.component.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (DawidKapka) [inline src/app/storify/create/create.component.ts:1]: unused import

File: src/app/storify/game/game.component.ts [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (DawidKapka) [inline src/app/storify/game/game.component.ts:1]: unused import

File: src/app/storify/join/join.component.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (DawidKapka) [inline src/app/storify/join/join.component.ts:1]: unused import

File: src/app/storify/shome/shome.component.ts [hunk 13]
```diff
@@ -0,0 +1,10 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (DawidKapka) [inline src/app/storify/shome/shome.component.ts:1]: unused import

File: src/app/storify/storify.component.ts [hunk 14]
```diff
@@ -0,0 +1,10 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (DawidKapka) [inline src/app/storify/storify.component.ts:1]: import

**Comment by fabianmueller7** [comment]
@Sili-Dev There is no preview for this :/

File: src/app/login/login.component.html [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+      <label class="text" for="email">Email:</label><br />
+      <app-textfield
+        type="email"
+        id="email"
+        name="email"
+        placeholder="email"
+        #email
+      ></app-textfield>
+    </div>
+    <br />
```
Comment (DawidKapka) [inline src/app/login/login.component.html:14]: please don't use <br> and instead actually add the correct margin

File: src/app/login/login.component.html [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+    <div class="label">
+      <label class="text" for="password">Password:</label><br />
+      <app-textfield
+        type="password"
+        id="password"
+        name="password"
+        placeholder="password"
+        #password
+      ></app-textfield>
+      <br />
```
Comment (DawidKapka) [inline src/app/login/login.component.html:24]: same here

File: src/app/login/login.component.scss [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+    white-space: normal;
+  }
+
+  .label {
+    align-content: start;
+    text-align: start;
+  }
+
+  .button {
+    margin-top: 2rm;
```
Comment (DawidKapka) [inline src/app/login/login.component.scss:26]: would be 36px

File: src/app/register/register.component.html [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+    <label class="text" for="username">Username:</label><br />
+    <app-textfield
+      type="text"
+      id="username"
+      name="username"
+      placeholder="username"
+      #username
+    ></app-textfield>
+  </div>
+  <br />
```
Comment (DawidKapka) [inline src/app/register/register.component.html:13]: same here

File: src/app/register/register.component.html [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+    <label class="text" for="email">Email:</label><br />
+    <app-textfield
+      type="email"
+      id="email"
+      name="email"
+      placeholder="email"
+      #email
+    ></app-textfield>
+  </div>
+  <br />
```
Comment (DawidKapka) [inline src/app/register/register.component.html:24]: and here

File: src/app/register/register.component.html [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+  <div class="label">
+    <label class="text" for="password">Password:</label><br />
+    <app-textfield
+      type="password"
+      id="password"
+      name="password"
+      placeholder="password"
+      #password
+    ></app-textfield>
+    <br />
```
Comment (DawidKapka) [inline src/app/register/register.component.html:34]: and here

File: src/app/register/register.component.scss [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+    white-space: normal;
+  }
+
+  .label {
+    align-content: start;
+    text-align: start;
+  }
+
+  .button {
+    margin-top: 2rem;
```
Comment (DawidKapka) [inline src/app/register/register.component.scss:26]: again, 36px

File: src/app/login/login.component.scss [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+  .subtitle {
+    font-size: 18px;
+    font-family: "Silkscreen", cursive;
+    margin: 1;
+    white-space: normal;
+    display: block;
+  }
+
+  .text {
+    font-size: 14px;
```
Comment (DawidKapka) [inline src/app/login/login.component.scss:14]: would be 16px

File: src/app/login/login.component.html [hunk 23]
```diff
@@ -0,0 +1,35 @@
...
+        type="email"
+        id="email"
+        name="email"
+        placeholder="email"
+        #email
+      ></app-textfield>
+    </div>
+    <br />
+    <div class="label">
+      <label class="text" for="password">Password:</label><br />
```
Comment (DawidKapka) [inline src/app/login/login.component.html:16]: same here with <br>

File: src/app/login/login.component.scss [hunk 24]
```diff
@@ -0,0 +1,40 @@
+.login {
+  display: grid;
+  grid-template-columns: 1/3 1/3;
+
+  .subtitle {
+    font-size: 18px;
+    font-family: "Silkscreen", cursive;
+    margin: 1;
```
Comment (DawidKapka) [inline src/app/login/login.component.scss:8]: ?

File: src/app/login/login.component.scss [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+    white-space: normal;
+    display: block;
+  }
+
+  .text {
+    font-size: 14px;
+    font-family: "Silkscreen", cursive;
+    margin: 1;
```
Comment (DawidKapka) [inline src/app/login/login.component.scss:16]: ?

File: src/app/login/login.component.scss [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  .button {
+    margin-top: 2rm;
+  }
+
+  .title {
+    font-size: 36px;
+    font-family: "Silkscreen", cursive;
+    margin: 1;
```
Comment (DawidKapka) [inline src/app/login/login.component.scss:32]: ?

File: src/app/register/register.component.scss [hunk 27]
```diff
@@ -0,0 +1,40 @@
+.login {
+  display: grid;
+  grid-template-columns: 1/3 1/3 1/3;
+
+  .subtitle {
+    font-size: 18px;
+    font-family: "Silkscreen", cursive;
+    margin: 1;
```
Comment (DawidKapka) [inline src/app/register/register.component.scss:8]: ?

File: src/app/register/register.component.scss [hunk 28]
```diff
@@ -0,0 +1,40 @@
...
+    white-space: normal;
+    display: block;
+  }
+
+  .text {
+    font-size: 14px;
```
Comment (DawidKapka) [inline src/app/register/register.component.scss:14]: 16px

File: src/app/register/register.component.scss [hunk 29]
```diff
@@ -0,0 +1,40 @@
...
+    font-family: "Silkscreen", cursive;
+    margin: 1;
```
Comment (DawidKapka) [inline src/app/register/register.component.scss:16]: ?

File: src/app/register/register.component.scss [hunk 30]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  .button {
+    margin-top: 2rem;
+  }
+
+  .title {
+    font-size: 36px;
+    font-family: "Silkscreen", cursive;
+    margin: 1;
```
Comment (DawidKapka) [inline src/app/register/register.component.scss:32]: ?

File: src/app/register/register.component.scss [hunk 31]
```diff
@@ -0,0 +1,40 @@
...
+    align-content: start;
+    text-align: start;
+  }
+
+  .button {
+    margin-top: 2rem;
+  }
+
+  .title {
+    font-size: 36px;
```
Comment (DawidKapka) [inline src/app/register/register.component.scss:30]: remove font-size and instead make title a <h3> element

File: src/app/login/login.component.scss [hunk 32]
```diff
@@ -0,0 +1,40 @@
...
+    align-content: start;
+    text-align: start;
+  }
+
+  .button {
+    margin-top: 2rm;
+  }
+
+  .title {
+    font-size: 36px;
```
Comment (DawidKapka) [inline src/app/login/login.component.scss:30]: remove font-size and instead make title a h3 element

File: src/app/login/login.component.html [hunk 33]
```diff
@@ -0,0 +1,33 @@
...
+        type="password"
+        id="password"
+        name="password"
+        placeholder="password"
+        #password
+      ></app-textfield>
+    </div>
+    <div class="pt-3 pl-0">
+      <app-button (click)="login(email.value, password.value)" size="large"
+        >Login</app-button
```
Comment (DawidKapka) [inline src/app/login/login.component.html:26]: wrong button size, would be medium

Comment (DawidKapka) [reply src/app/login/login.component.html:26]: also, wrong padding

**Review by DawidKapka — APPROVED** [review]

**Pull request merged**